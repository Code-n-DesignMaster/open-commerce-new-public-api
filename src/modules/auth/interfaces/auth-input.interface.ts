export interface IAuthInput {
  email?: string;
  birthdate?: string;
}
