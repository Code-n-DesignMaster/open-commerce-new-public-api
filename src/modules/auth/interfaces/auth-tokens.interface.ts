export interface IAuthTokens {
  tokens: {
    accessToken: string;
    refreshToken: string;
  };
}
