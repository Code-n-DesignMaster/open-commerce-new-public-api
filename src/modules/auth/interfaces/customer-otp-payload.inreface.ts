export interface ICustomerOtpPayload {
  session: string;
  success: boolean;
  sub: string;
}
