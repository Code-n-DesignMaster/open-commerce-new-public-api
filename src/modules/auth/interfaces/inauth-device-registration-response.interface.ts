export interface IInAuthDeviceRegistrationResponse {
  readonly deviceResponse: string;
}
