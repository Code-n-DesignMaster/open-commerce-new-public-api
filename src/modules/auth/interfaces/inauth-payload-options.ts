import { IAuth } from '../../common/interfaces/auth.interface';

export interface IInAuthPayloadOptions {
  auth?: IAuth;
  deviceId: string;
  deviceIpAddress: string;
  transactionId: string;
  inAuthLogPayload: string;
}
