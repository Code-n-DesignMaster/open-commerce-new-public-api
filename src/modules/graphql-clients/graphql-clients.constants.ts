export const CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const TRANSACTION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'TRANSACTION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN';

export const CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const TRANSACTION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'TRANSACTION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN';

export const GRAPHQL_SERVICE_DIRECTORY_TOKEN =
  'GRAPHQL_SERVICE_DIRECTORY_TOKEN';
