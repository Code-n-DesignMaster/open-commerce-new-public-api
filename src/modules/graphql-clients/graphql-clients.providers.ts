import { Provider } from '@nestjs/common';
import {
  CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN,
  NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
  MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN,
  CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN,
  NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
  MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN,
  DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN,
  DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN,
  API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
  API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
  PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN,
  PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN,
  AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN,
  AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN,
  GRAPHQL_SERVICE_DIRECTORY_TOKEN,
} from './graphql-clients.constants';
import {
  CustomerServiceGraphqlClient,
  NotificationServiceGraphqlClient,
  PassManagementServiceGraphqlClient,
  DaveAndBustersServiceGraphqlClient,
  ApiVerificationServiceGraphqlClient,
  PaymentServiceGraphqlClient,
  AuthServiceGraphqlClient,
  GraphqlServiceDirectory,
  SERVICE_NAME,
} from '@open-commerce/db-internal-service-client';
import { CustomerGraphqlClientService } from './clients/customer-graphql-client.service';
import { NotificationGraphqlClientService } from './clients/notification-graphql-client.service';
import { MobilePassGraphqlClientService } from './clients/mobile-pass-graphql-client.service';
import { DaveAndBustersGraphqlClientService } from './clients/dave-and-busters-graphql-client.service';
import { ApiVerificationGraphqlClientService } from './clients/api-verification-graphql-client.service';
import { PaymentGraphqlClientService } from './clients/payment-graphql-client.service';
import { AuthGraphqlClientService } from './clients/auth-graphql-client.service';

export const graphqlClientsProviders: Provider[] = [
  {
    provide: GRAPHQL_SERVICE_DIRECTORY_TOKEN,
    useFactory: () =>
      new GraphqlServiceDirectory([
        SERVICE_NAME.PASS_MANAGEMENT,
        SERVICE_NAME.CUSTOMER,
      ]),
  },
  // NOTE: These are the internal service client instances from
  // internal service client NPM.
  {
    provide: CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: CustomerServiceGraphqlClient,
  },
  {
    provide: NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: NotificationServiceGraphqlClient,
  },
  {
    provide: MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: PassManagementServiceGraphqlClient,
  },
  {
    provide: DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: DaveAndBustersServiceGraphqlClient,
  },
  {
    provide: API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: ApiVerificationServiceGraphqlClient,
  },
  {
    provide: PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: PaymentServiceGraphqlClient,
  },
  {
    provide: AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: AuthServiceGraphqlClient,
  },
  //////////////////////////////////////////////

  // NOTE: These are helper abstractions that contain code to
  // build the operations executed by the internal service clients above.
  {
    provide: CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: CustomerGraphqlClientService,
  },
  {
    provide: NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: NotificationGraphqlClientService,
  },
  {
    provide: MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: MobilePassGraphqlClientService,
  },
  {
    provide: DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: DaveAndBustersGraphqlClientService,
  },
  {
    provide: API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: ApiVerificationGraphqlClientService,
  },
  {
    provide: PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: PaymentGraphqlClientService,
  },
  {
    provide: AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: AuthGraphqlClientService,
  },
  ///////////////////////////////////
];
