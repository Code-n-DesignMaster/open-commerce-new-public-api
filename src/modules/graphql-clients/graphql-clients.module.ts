import { Module, Global } from '@nestjs/common';
import { graphqlClientsProviders } from './graphql-clients.providers';

@Global()
@Module({
  providers: graphqlClientsProviders,
  exports: graphqlClientsProviders,
})
export class GraphqlClientsModule {}
