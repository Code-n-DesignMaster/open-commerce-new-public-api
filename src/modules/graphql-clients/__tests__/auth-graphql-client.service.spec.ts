import { AuthGraphqlClientService } from '../clients/auth-graphql-client.service';
import { Test } from '@nestjs/testing';
import { AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN } from '../graphql-clients.constants';
import { LoggerService } from '@open-commerce/nestjs-logger';
import { mockGraphqlClientExecuteQuery } from './mock-graphql-client-execute-query';
import { IInAuthPayloadOptions } from '../../auth/interfaces/inauth-payload-options';
import { InAuthDeviceRegistrationRequestInputDto } from '../../auth/dtos/inauth-device-registration-request-input.dto';
import { InauthTrustScoreValueInputDto } from '../../auth/dtos/inauth-trust-score-value-input.dto';
import { CustomerOtpRequestDto } from '../../auth/dtos/customer-otp-request.dto';
import { OTPChannel } from '../../auth/otp-channel.enum';
import { IAuthTokens } from '../../auth/interfaces/auth-tokens.interface';
import { CustomerLoginByOtpDto } from '../../auth/dtos/customer-login-by-otp.dto';

const mockLogger = {
  debug: jest.fn(),
  warn: jest.fn(),
  info: jest.fn(),
  error: jest.fn(),
  log: jest.fn(),
};

const mockGraphqlClient = {
  executeQuery: jest.fn(),
};

describe('AuthGraphqlClientService', () => {
  let service: AuthGraphqlClientService;

  beforeAll(async () => {
    const module = await Test.createTestingModule({
      providers: [
        AuthGraphqlClientService,
        {
          provide: LoggerService,
          useValue: mockLogger,
        },
        {
          provide: AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN,
          useValue: mockGraphqlClient,
        },
      ],
    }).compile();

    service = module.get(AuthGraphqlClientService);
  });

  describe('customerUsernameChange', () => {
    const newUsername = 'testNewUsername';
    const sub = 'testSub';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerUsernameChange',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerUsernameChange(newUsername, sub);
    });
  });

  describe('customerUsernameChangeVerify', () => {
    const otp = {
      otp: 'testOtp',
      session: 'testSession',
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'customerUsernameChangeVerify',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.customerUsernameChangeVerify(otp);
    });
  });

  describe('inAuthPayloadValidate', () => {
    const input: IInAuthPayloadOptions = {
      auth: {
        sub: 'testSub',
        accessToken: 'testAccessToken',
        expiresIn: 655321,
      },
      deviceId: 'testDeviceId',
      deviceIpAddress: 'testIpAddress',
      inAuthLogPayload: 'testInAuthPayload',
      transactionId: 'testTransactionId',
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('inAuthPayloadValidate', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.inAuthPayloadValidate(input);
    });
  });

  describe('inAuthMobileDeviceRegister', () => {
    const input: InAuthDeviceRegistrationRequestInputDto = {
      deviceId: 'testDeviceId',
      payload: 'testInAuthPayload',
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('inauthDeviceRegister', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.inAuthMobileDeviceRegister(input);
    });
  });

  describe('inauthDeviceTrustScoreThresholdUpdate', () => {
    const input: InauthTrustScoreValueInputDto = {
      value: 42,
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'inauthDeviceTrustScoreThresholdUpdate',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.inauthDeviceTrustScoreThresholdUpdate(input);
    });
  });

  describe('inauthCustomerToDeviceConnect', () => {
    const input = {
      deviceId: 'testDeviceId',
      customerId: 42,
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'inauthCustomerToDeviceConnect',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.inauthCustomerToDeviceConnect(input);
    });
  });

  describe('inauthBlacklistedDeviceUnblockedStatusSet', () => {
    const deviceId = 'testDeviceId';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'inauthBlacklistedDeviceUnblockedStatusSet',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.inauthBlacklistedDeviceUnblockedStatusSet(deviceId, false);
    });
  });

  describe('inauthDeviceBlacklistRemove', () => {
    const deviceId = 'testDeviceId';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'inauthDeviceBlacklistRemove',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.inauthDeviceBlacklistRemove(deviceId);
    });
  });

  describe('inAuthGetSigfiles', () => {
    const input = {
      deviceId: 'testDeviceId',
      deviceType: 'testDeviceType',
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('inauthGetSigfiles', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.inAuthGetSigfiles(input);
    });
  });

  describe('customerOTPRequest', () => {
    const input: CustomerOtpRequestDto = {
      identifier: 'testIdentifier',
      channel: OTPChannel.EMAIL,
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerOTPRequest', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerOTPRequest(input);
    });
  });

  describe('updateUserAttrs', () => {
    const input = {
      name: {
        first: 'testFirstName',
      },
    };
    const accessToken = 'testAccessToken';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('updateUserAttrs', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.updateUserAttrs(input, accessToken);
    });
  });

  describe('logout', () => {
    const accessToken = 'testAccessToken';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('logout', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.logout(accessToken);
    });
  });

  describe('getAccessTokenByRefreshToken', () => {
    const input: IAuthTokens = {
      tokens: {
        accessToken: 'testAccessToken',
        refreshToken: 'testRefreshToken',
      },
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'getAccessTokenByRefreshToken',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.getAccessTokenByRefreshToken(input);
    });
  });

  describe('customerLoginByOTP', () => {
    const deviceId = 'testDeviceId';

    const input: CustomerLoginByOtpDto = {
      otp: 'testOtp',
      session: 'testSession',
    };

    beforeAll(() => {
      mockGraphqlClientExecuteQuery('customerLoginByOTP', mockGraphqlClient);
    });
    it('should match snapshot', async () => {
      await service.customerLoginByOTP(input, deviceId);
    });
  });
});
