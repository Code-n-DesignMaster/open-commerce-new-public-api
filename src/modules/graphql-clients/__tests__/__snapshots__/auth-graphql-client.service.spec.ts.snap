// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthGraphqlClientService customerLoginByOTP should match snapshot: for operation 1`] = `
"mutation customerLoginByOTP($input: OTP!) {
 customerLoginByOTP(input: $input) {

  ... on CustomerOTPPayload {
   session
   success
  }
  ... on AuthTokens {
   tokens {
    accessToken
    refreshToken
   }
  }
 }
}"
`;

exports[`AuthGraphqlClientService customerLoginByOTP should match snapshot: for parameters 1`] = `
Object {
  "deviceId": "testDeviceId",
  "input": Object {
    "otp": "testOtp",
    "session": "testSession",
  },
}
`;

exports[`AuthGraphqlClientService customerOTPRequest should match snapshot: for operation 1`] = `
"mutation customerOTPRequest($input: CustomerOTPRequest!) {
 customerOTPRequest(input: $input) {

  session
  success
  sub
 }
}"
`;

exports[`AuthGraphqlClientService customerOTPRequest should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "channel": "EMAIL",
    "identifier": "testIdentifier",
  },
}
`;

exports[`AuthGraphqlClientService customerUsernameChange should match snapshot: for operation 1`] = `
"mutation customerUsernameChange($newUsername: String, $sub: String) {
 customerUsernameChange(newUsername: $newUsername, sub: $sub)
}"
`;

exports[`AuthGraphqlClientService customerUsernameChange should match snapshot: for parameters 1`] = `
Object {
  "newUsername": "testNewUsername",
  "sub": "testSub",
}
`;

exports[`AuthGraphqlClientService customerUsernameChangeVerify should match snapshot: for operation 1`] = `
"mutation customerUsernameChangeVerify($otp: OTP) {
 customerUsernameChangeVerify(otp: $otp)
}"
`;

exports[`AuthGraphqlClientService customerUsernameChangeVerify should match snapshot: for parameters 1`] = `
Object {
  "otp": Object {
    "otp": "testOtp",
    "session": "testSession",
  },
}
`;

exports[`AuthGraphqlClientService getAccessTokenByRefreshToken should match snapshot: for operation 1`] = `
"mutation getAccessTokenByRefreshToken($input: AccessTokenByRefreshTokenInputType!) {
 getAccessTokenByRefreshToken(input: $input) {

  tokens {
   accessToken
   refreshToken
  }
 }
}"
`;

exports[`AuthGraphqlClientService getAccessTokenByRefreshToken should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "tokens": Object {
      "accessToken": "testAccessToken",
      "refreshToken": "testRefreshToken",
    },
  },
}
`;

exports[`AuthGraphqlClientService inAuthGetSigfiles should match snapshot: for operation 1`] = `
"query inauthGetSigfiles($input: InAuthGetSigfilesRequestInput!) {
 inauthGetSigfiles(input: $input)
}"
`;

exports[`AuthGraphqlClientService inAuthGetSigfiles should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "deviceId": "testDeviceId",
    "deviceType": "testDeviceType",
  },
}
`;

exports[`AuthGraphqlClientService inAuthMobileDeviceRegister should match snapshot: for operation 1`] = `
"mutation inauthDeviceRegister($input: InAuthDeviceRegistrationRequestInput!) {
 inauthDeviceRegister(input: $input) {
  deviceResponse
 }
}"
`;

exports[`AuthGraphqlClientService inAuthMobileDeviceRegister should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "deviceId": "testDeviceId",
    "payload": "testInAuthPayload",
  },
}
`;

exports[`AuthGraphqlClientService inAuthPayloadValidate should match snapshot: for operation 1`] = `
"mutation inAuthPayloadValidate($input: InAuthPayloadInput!) {
 inAuthPayloadValidate(input: $input) {
  success
 }
}"
`;

exports[`AuthGraphqlClientService inAuthPayloadValidate should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "auth": Object {
      "accessToken": "testAccessToken",
      "expiresIn": 655321,
      "sub": "testSub",
    },
    "deviceId": "testDeviceId",
    "deviceIpAddress": "testIpAddress",
    "inAuthLogPayload": "testInAuthPayload",
    "transactionId": "testTransactionId",
  },
}
`;

exports[`AuthGraphqlClientService inauthBlacklistedDeviceUnblockedStatusSet should match snapshot: for operation 1`] = `
"mutation inauthBlacklistedDeviceUnblockedStatusSet($deviceId: ID!, $unblockedStatus: Boolean!) {
 inauthBlacklistedDeviceUnblockedStatusSet(deviceId: $deviceId, unblockedStatus: $unblockedStatus)
}"
`;

exports[`AuthGraphqlClientService inauthBlacklistedDeviceUnblockedStatusSet should match snapshot: for parameters 1`] = `
Object {
  "deviceId": "testDeviceId",
  "unblockedStatus": false,
}
`;

exports[`AuthGraphqlClientService inauthCustomerToDeviceConnect should match snapshot: for operation 1`] = `
"mutation inauthCustomerToDeviceConnect($input: InauthCustomerToDeviceConnectInput!) {
 inauthCustomerToDeviceConnect(input: $input)
}"
`;

exports[`AuthGraphqlClientService inauthCustomerToDeviceConnect should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "customerId": 42,
    "deviceId": "testDeviceId",
  },
}
`;

exports[`AuthGraphqlClientService inauthDeviceBlacklistRemove should match snapshot: for operation 1`] = `
"mutation inauthDeviceBlacklistRemove($deviceId: ID!) {
 inauthDeviceBlacklistRemove(deviceId: $deviceId)
}"
`;

exports[`AuthGraphqlClientService inauthDeviceBlacklistRemove should match snapshot: for parameters 1`] = `
Object {
  "deviceId": "testDeviceId",
}
`;

exports[`AuthGraphqlClientService inauthDeviceTrustScoreThresholdUpdate should match snapshot: for operation 1`] = `
"mutation inauthDeviceTrustScoreThresholdUpdate($input: InauthTrustScoreValueInput!) {
 inauthDeviceTrustScoreThresholdUpdate(input: $input)
}"
`;

exports[`AuthGraphqlClientService inauthDeviceTrustScoreThresholdUpdate should match snapshot: for parameters 1`] = `
Object {
  "input": Object {
    "value": 42,
  },
}
`;

exports[`AuthGraphqlClientService logout should match snapshot: for operation 1`] = `
"mutation logout($accessToken: String) {
 logout(accessToken: $accessToken)
}"
`;

exports[`AuthGraphqlClientService logout should match snapshot: for parameters 1`] = `
Object {
  "accessToken": "testAccessToken",
}
`;

exports[`AuthGraphqlClientService updateUserAttrs should match snapshot: for operation 1`] = `
"mutation updateUserAttrs($input: UserAttributes, $accessToken: String) {
 updateUserAttrs(input: $input, accessToken: $accessToken)
}"
`;

exports[`AuthGraphqlClientService updateUserAttrs should match snapshot: for parameters 1`] = `
Object {
  "accessToken": "testAccessToken",
  "input": Object {
    "name": Object {
      "first": "testFirstName",
    },
  },
}
`;
